src/model.ts

```diff
@@ -27,20 +22,35 @@ export class GitExtension implements IGitExtension, IDisposable {
         console.error(`Fail to get the server root path.\n${reason}`);
       });
 
-    const refreshInterval = DEFAULT_REFRESH_INTERVAL;
-
     // Start watching the repository
     this._poll = new Poll({
       factory: () => this._refreshStatus(),
       frequency: {
-        interval: refreshInterval,
+        interval: this._refreshInterval,
         backoff: true,
         max: 300 * 1000
       },
       standby: 'when-hidden'
     });
   }
 
+  /**
+   * Number of milliseconds between polling the file system for changes.
+   */
+  public get refreshInterval(): number {
+    return this._refreshInterval;
+  }
+
+  public set refreshInterval(v: number) {
```

Comment:
Is there a reasonable way that we could make the `set` private or protected (ie called only in constructor and on settings change)?

Essentially either: 

a) `set` has to be private and carefully handled, or 
b) designed in such a way as to ensure that it can never cause a desyncing with `settings`

Two possible designs for the latter are to:
- pass `settings` in somehow and use its `set` method (see this [bit of ThemeManager](https://github.com/jupyterlab/jupyterlab/blob/78e7bf2d6848fe22f2b59200c832e60f83b6ad22/packages/apputils/src/thememanager.ts#L195-L263) for various examples)
- leave the refresh logic in place put hoist the polling logic out of `GitExtension` and into `index.ts` where the `settings` is first materialized

Both make sense, but it also makes sense for the `GitExtension` model to not be tightly coupled to the concept of jlab's global application settings, so I'm leaning towards hoist.

Reply:
Hoist ended being a bit messy, as the Git model currently has logic for managing the poll. So I opted for the former: linking the model with the settings.